src/Ocelot/DownstreamRouteFinder/UrlMatcher/UrlPathPlaceholderNameAndValueFinder.cs

```diff
@@ -67,10 +67,12 @@ public Response<List<PlaceholderNameAndValue>> Find(string path, string query, s
     private static List<Group> FindGroups(string path, string query, string template)
     {
         template = EscapeExceptBraces(template);
-        var regexPattern = $"^{RegexPlaceholders().Replace(template, match => $"(?<{match.Groups[1].Value}>[^&]*)")}";
+        var regexPattern = GenerateRegexPattern(template);
```

Comment:
Seems this is dynamic `Regex`, so the pattern is dynamically generated, but the previous version aka `RegexPlaceholders()` returns static object.
Well... is it possible to convert somehow to static object to improve the performance?